test-tools/src/main/java/org/terracotta/utilities/test/Diagnostics.java

```diff
@@ -389,6 +403,52 @@ public static MaxDirectMemoryInfo getMaxDirectMemoryInfo() {
     return MaxDirectMemoryInfoHelper.INSTANCE;
   }
 
+  /**
+   * Handles using the JVM-appropriate method to obtain the id for a {@code Thread}.
+   */
+  private static class ThreadId {
```

Comment:
I could see moving this nested class elsewhere to make available to non-test code.